starter-code/memory.js

```diff
@@ -31,32 +31,101 @@ var MemoryGame = function() {
     this.selectedCards = [];
     this.pairsClicked = 0;
     this.correctPairs = 0;
+    this.matchedCards = [];
 };
 
+MemoryGame.prototype._shuffleCards = function() {
+  this.cards=_.shuffle(this.cards);
+};
+
+MemoryGame.prototype.selectCard = function(clickedCard){
+  if(this.selectedCards.length === 0){
+    this.selectedCards.push(clickedCard);
+  } else if(this.selectedCards[0] === clickedCard){
+      this.correctPairs++;
+      $("#pairs_guessed").text(this.correctPairs);
+      this.pairsClicked++;
+      $("#pairs_clicked").text(this.pairsClicked);
+      this.matchedCards.push(clickedCard);
+      this.matchedCards.push(this.selectedCards[0]);
+
+      console.log("this.matchedCards[0]:", this.matchedCards[0]);
+      console.log("this.matchedCards:", this.matchedCards);
+
+
+        var name = this.matchedCards[name];
+        $('.front[name=' + name + ']').parent().addClass('match');
+
+
+
+      _.remove(this.selectedCards);
+      //console.log(clickedCard);
+      //console.log($(this).attr("name"));
+
+      //$("div[name='flash']").removeClass(".front");
+
+
+  } else if(this.selectedCards[0] !== clickedCard){
+    this.pairsClicked++;
+    $("#pairs_clicked").text(this.pairsClicked);
+    _.remove(this.selectedCards);
+    $(".card").not('.match').children('.front').delay(1000).hide(0);
```

Comment:
The reason you are hiding all the cards after every bad guess is due to the code not distinguishing between cards that have been correctly guessed and all the rest which are still "in play" (but I assume you know this already :) 

There are several ways to solve this, one of which is to add a "available" class to all cards and remove it once they have been guessed. Another could be holding a property in memoryGame to remember these cards. Implementing this naturally takes some time, and it's fine if you didn't manage to do it during the day. We will practice these concepts more this week.